reflex-algos/src/main/python/mlops/parallelm/mlops/time_capture/plot_functions.py

```diff
@@ -284,20 +357,56 @@ def align_bins(self, df):
+            rounded_bins = bins
+            for keys_origin_index in range(0, num_of_bins):
+                if ' to ' in bins[keys_origin_index]:
+                    bins_vect = bins[keys_origin_index].split(' to ')
+                    rounded_bins_vect = bins_vect
+                    for index_bin in range(0,2):
+                        if 'inf' not in bins_vect[index_bin]:
+                            rounded_bins_vect[index_bin] = str(round(float(bins_vect[index_bin]), 4))
+                    rounded_bins[keys_origin_index] = rounded_bins_vect[0] + ' to ' + rounded_bins_vect[1]
 
             for location_index in range(0, num_graphs):
                 df_values = []
                 for keys_origin_index in range(0, num_of_bins):
                     append_val = 0
-                    for key_index in range(0, len(df_keys.loc[location_index])):
-                        if bins[keys_origin_index] == \
-                                df_keys.loc[location_index][key_index]:
-                            append_val = df["values"].loc[location_index][key_index]
+                    if ' to ' in bins[keys_origin_index]:
+                        bins1 = df_keys.loc[location_index][keys_origin_index]
+                        bins1_vect = bins1.split(' to ')
+                        rounded_bins1_vect = bins1_vect
+                        for index_bin in range(0,2):
+                            if 'inf' not in bins1_vect[index_bin]:
+                                rounded_bins1_vect[index_bin] = str(round(float(bins1_vect[index_bin]), 4))
+                        rounded_bins1 = rounded_bins1_vect[0] + ' to ' + rounded_bins1_vect[1]
+                        if rounded_bins[keys_origin_index] == \
+                                rounded_bins1:
+                            append_val = df["values"].loc[location_index][keys_origin_index]
+                    else:
+                        for key_index in range(0, len(df_keys.loc[location_index])):
+                            if bins[keys_origin_index] == \
+                                    df_keys.loc[location_index][key_index]:
+                                append_val = df["values"].loc[location_index][key_index]
                     df_values.append(append_val)
 
                 td_matrix_loc[location_index, :] = df_values
 
-                name_pipeline_vect = df["FileName"].loc[location_index].split('-')
-                name_pipeline = name_pipeline_vect[len(name_pipeline_vect) - 4]
+                name_pipeline_vect = df["FileName"].loc[location_index].split('-Instance-')
+                name_pipeline = (name_pipeline_vect[1].split('-Agent-'))[0]
                 name_pipeline1.append(name_pipeline)
             return name_pipeline1, td_matrix_loc
+
+    def file_to_node(self, filename):
+        """
+        get node name from file name:
+
+        :param self:
+        :param filename:
+        :return node_name:
+        """
+        node_info = self._mtc.get_nodes()
```

Comment:
do we think we need to ensure the timeline capture is formatted well - as in the user hasnt opened it and messed around with it?

Reply:
yea I guess it make sense on a longer run, once we stabilize. but I think you will get first level of it in the parser as it is expecting a certain format to parse and you will get an exception if not.